test/common/sink_test.cc

```diff
@@ -0,0 +1,138 @@
+  ASSERT_EQ(status_or_execution_responses.ok(), false);
+  EXPECT_EQ(status_or_execution_responses.status().code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(status_or_execution_responses.status().message(),
+            "'' is not a guid: bad string length.");
+}
+
+TYPED_TEST(TypedSinkTest, Append) {
+  TypeParam sink;
+  nighthawk::client::ExecutionResponse result_to_store;
+  *(result_to_store.mutable_execution_id()) = this->executionIdForTest();
+  absl::Status status = sink.StoreExecutionResultPiece(result_to_store);
+  ASSERT_EQ(status.ok(), true);
+  status = sink.StoreExecutionResultPiece(result_to_store);
+  ASSERT_EQ(status.ok(), true);
+  const auto status_or_execution_responses = sink.LoadExecutionResult(this->executionIdForTest());
+  EXPECT_EQ(status_or_execution_responses.value().size(), 2);
+}
+
+// As of today, we constrain execution id to a guid. This way the file sink implementation
+// ensures that it can safely use it to create directories. In the future, other sinks may not
+// have to worry about such things. In that case it makes sense to add a validation call
+// to the sink interface to make this implementation specific, and make the tests below
+// implementation specific too.
+TYPED_TEST(TypedSinkTest, BadGuidShortString) {
+  TypeParam sink;
+  const auto status_or_execution_responses =
+      sink.LoadExecutionResult("14e75b2a-3e31-4a62-9279-add1e54091f");
+  ASSERT_EQ(status_or_execution_responses.ok(), false);
+  EXPECT_EQ(status_or_execution_responses.status().code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(status_or_execution_responses.status().message(),
+            "'14e75b2a-3e31-4a62-9279-add1e54091f' is not a guid: bad string length.");
+}
+
+TYPED_TEST(TypedSinkTest, BadGuidBadDashPlacement) {
+  TypeParam sink;
+  const auto status_or_execution_responses =
+      sink.LoadExecutionResult("14e75b2a3-e31-4a62-9279-add1e54091f9");
+  ASSERT_EQ(status_or_execution_responses.ok(), false);
+  EXPECT_EQ(status_or_execution_responses.status().code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(status_or_execution_responses.status().message(),
+            "'14e75b2a3-e31-4a62-9279-add1e54091f9' is not a guid: expectations around '-' "
+            "positions not met.");
+}
+
+TYPED_TEST(TypedSinkTest, BadGuidInvalidCharacter) {
+  TypeParam sink;
+  const auto status_or_execution_responses =
+      sink.LoadExecutionResult("14e75b2a-3e31-4x62-9279-add1e54091f9");
+  ASSERT_EQ(status_or_execution_responses.ok(), false);
+  EXPECT_EQ(status_or_execution_responses.status().code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(
+      status_or_execution_responses.status().message(),
+      "'14e75b2a-3e31-4x62-9279-add1e54091f9' is not a guid: unexpected character encountered.");
+}
+
+TEST(FileSinkTest, CorruptedFile) {
+  FileSinkImpl sink;
+  const std::string execution_id = "14e75b2a-3e31-4162-9279-add1e54091f9";
```

Comment:
Any reason not to use `this->executionIdForTest()` here?

Reply:
Well, this test is different from the others as it doesn't have a fixture. (it's not part of the TEST_P ones)

Reply:
Ohh I see. Do we need to clean up the directory separately here?